app/models/conversion_host.rb

```diff
@@ -4,4 +4,104 @@ class ConversionHost < ApplicationRecord
   acts_as_miq_taggable
 
   belongs_to :resource, :polymorphic => true
+  has_many :service_template_transformation_plan_tasks, :dependent => :nullify
+
+  def active_tasks
+    service_template_transformation_plan_tasks { active }
+  end
+
+  def eligible?
+    return true if concurrent_transformation_limit.nil?
+    active_tasks.size < concurrent_transformation_limit.to_i
+  end
+
+  def source_transport_method
+    return 'vddk' if vddk_transport_supported
+    return 'ssh' if ssh_transport_supported
+  end
+
+  def conversion_options(task)
+    source_vm = task.source
+    source_ems = source_vm.ext_management_system
+    source_cluster = source_vm.ems_cluster
+    source_storage = source_vm.hardware.disks.select { |d| d.device_type == 'disk' }.first.storage
+
+    destination_cluster = task.transformation_destination(source_cluster)
+    destination_storage = task.transformation_destination(source_storage)
+    destination_ems = destination_cluster.ext_management_system
+
+    source_provider_options = send(
+      "conversion_options_source_provider_#{source_ems.emstype}_#{source_transport_method}",
+      source_vm,
+      source_storage
+    )
+    destination_provider_options = send(
+      "conversion_options_destination_provider_#{destination_ems.emstype}",
+      task,
+      destination_ems,
+      destination_cluster,
+      destination_storage
+    )
+    options = {
+      :source_disks     => task.source_disks.map { |disk| disk[:path] },
+      :network_mappings => task.network_mappings
+    }
+    options.merge(source_provider_options).merge(destination_provider_options)
+  end
+
+  def conversion_options_source_provider_vmwarews_vddk(vm, _storage)
```

Comment:
@fdupont-redhat should these below methods be on the conversion_task model?  What from the conversion_host do they use?

Reply:
Indeed, they should be in the task... And that also answers https://github.com/ManageIQ/manageiq/pull/18033#discussion_r222338708

Reply:
:+1: awesome, if there is anything that the task needs from the conversion_host we can delegate those to the conversion_host as @blomquisg said above